src/main/java/io/fabric8/maven/docker/AbstractBuildSupportMojo.java

```diff
@@ -60,7 +65,17 @@ protected void buildImage(ServiceHub hub, ImageConfiguration imageConfig)
         autoPullBaseImage(hub, imageConfig);
 
         MojoParameters params = createMojoParameters();
-        hub.getBuildService().buildImage(imageConfig, params, checkForNocache(imageConfig));
+        hub.getBuildService().buildImage(imageConfig, params, checkForNocache(imageConfig), addBuildArgs());
+    }
+
+    private Map<String, String> addBuildArgs() {
+        HashMap<String, String> args = new HashMap<>();
+        String[] argPairs = buildArgs.split(",");
```

Comment:
Also wonder whether we need a distinction for individual images here since a build can contain many images (e.g. `docker.build.<image|alias>.arg.<varname>`

To make thinks shorter what about:

* `docker.arg.<varname>` --> Used for all images
* `docker.iarg.<image>.<varname>` --> Used for image `<image>`.